siatest/renter/renter_test.go

```diff
@@ -1203,3 +1204,77 @@ func TestRedundancyReporting(t *testing.T) {
 		t.Fatal("Redundancy is not increasing")
 	}
 }
+
+// testClearDownloadHistory makes sure that the download history is
+// properly cleared when called through the API
+func testClearDownloadHistory(t *testing.T, tg *siatest.TestGroup) {
+	// Grab the first of the group's renters
+	r := tg.Renters()[0]
+
+	rdg, err := r.RenterDownloadsGet()
+	if err != nil {
+		t.Fatal("Could not get download history:", err)
+	}
+	numDownloads := len(rdg.Downloads)
+	if numDownloads == 0 {
```

Comment:
I'd declare a variable here that indicates how many downloads you want to see in the history. Probably 5 is fine. If there are fewer than 5 you add enough downloads to fill the gap.
I like the idea of reusing existing downloads.